src/channels.rs

```diff
@@ -118,6 +118,38 @@ where
             .await
             .map_err(Error::I2C)
     }
+
+    /// Set the PWM control registers for each channel at once.
+    ///
+    /// This allows to set all `on` and `off` counter values, as well as the
+    /// full-on and full-off bit in a single I2C transaction.
+    /// The index of the value in the array corresponds to the channel: 0-15.
+    ///
+    /// See section 7.3.3 "LED output and PWM control" of the datasheet for
+    /// further details.
+    pub async fn set_all_channels(
+        &mut self,
+        values: &[ChannelOnOffControl; 16],
+    ) -> Result<(), Error<E>> {
+        let mut data = [0; 65];
+        data[0] = Register::C0_ON_L;
+        for (i, channel_value) in values.iter().enumerate() {
+            if channel_value.on > 4095 || channel_value.off > 4095 {
+                return Err(Error::InvalidInputData);
+            }
+            data[i * 4 + 1] = channel_value.on as u8;
+            data[i * 4 + 2] =
+                (channel_value.on >> 8) as u8 | (0b0001_0000 * channel_value.full_on as u8);
```

Comment:
Could you create a local constant like `const FULL_ON_OFF: u8 = 0b0001_0000;` for this?